modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/AMQPUtils.java

```diff
@@ -215,7 +215,12 @@ public static AndesMessageMetadata convertAMQMessageToAndesMetadata(AMQMessage a
         amqMetadata.writeToBuffer(0, buf);
 
         AndesMessageMetadata metadata = new AndesMessageMetadata(amqMessage.getMessageId(),underlying,true);
-        metadata.setExpirationTime(amqMessage.getExpiration());
+        //In andes message set the expiry time as the addition of arrival time with the ttl
+        long messageExpiryTime = 0L;
+        if(amqMessage.getExpiration() != 0L){
```

Comment:
The usual practice is that we have the constant on the left side of the comparison